registry/client/repository.go

```diff
@@ -307,7 +307,7 @@ func (t *tags) Get(ctx context.Context, tag string) (distribution.Descriptor, er
 	}
 
 	newRequest := func(method string) (*http.Response, error) {
-		req, err := http.NewRequest(method, u, nil)
+		req, err := newRequestWithContext(ctx, method, u, nil)
```

Comment:
Now the method can be used from the `http` package

Reply:
I don't think I follow

Reply:
https://golang.org/pkg/net/http/#NewRequestWithContext

Also for the other cases, the helper functions shouldn't be necessary after this function is used

Reply:
Gotcha. The code is now using `http.NewRequestWithContext()`.

I also removed the auxiliary functions but I think it looked cleaner with them.